core/controllers/classifier_test.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the controllers that communicate with VM for training
+classifiers."""
+
+import hashlib
+import hmac
+import json
+import os
+
+from core.domain import classifier_services
+from core.domain import exp_services
+from core.tests import test_utils
+import feconf
+
+
+def generate_signature(data):
+    """Generates digital signature for given data."""
+    msg = json.dumps(data, sort_keys=True)
+    key = feconf.DEFAULT_VM_SHARED_SECRET
+    return hmac.new(key, msg, digestmod=hashlib.sha256).hexdigest()
+
+
+class TrainedClassifierHandlerTest(test_utils.GenericTestBase):
+    """Test the handler for storing job result of training job."""
+
+    def setUp(self):
+        super(TrainedClassifierHandlerTest, self).setUp()
+
+        self.exp_id = 'exp_id1'
+        self.title = 'Testing Classifier storing'
+        self.category = 'Test'
+        yaml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                 '../tests/data/string_classifier_test.yaml')
```

Comment:
Break up the path into chunks and use os.path.join. Otherwise this will break on non-unix systems. (But with feconf.TESTS_DATA_DIR you may not even need os.path.join -- look elsewhere in the codebase to see where the constant is used.)